src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/QinQFlowSpec.groovy

```diff
@@ -81,129 +89,115 @@ class QinQFlowSpec extends HealthCheckSpecification {
-        [qinqFlow.flowId, vlanFlow.id].each {
-            northbound.validateFlow(it).each { assert it.asExpected }
+        [qinqFlow, vlanFlow].each {
+            it.validateAndCollectDiscrepancies().isEmpty()
         }
 
         and: "Involved switches pass switch validation"
-        def involvedSwitchesFlow2 = pathHelper.getInvolvedSwitches(pathHelper.convert(northbound.getFlowPath(vlanFlow.id)))
-        def involvedSwitchesforBothFlows = (involvedSwitchesFlow1 + involvedSwitchesFlow2).unique { it.dpId }
-        switchHelper.synchronizeAndCollectFixedDiscrepancies(involvedSwitchesforBothFlows*.getDpId()).isEmpty()
+        def involvedSwitchesFlow2 = vlanFlow.retrieveAllEntityPaths().getInvolvedSwitches()
+        def involvedSwitchesforBothFlows = (involvedSwitchesFlow1 + involvedSwitchesFlow2).unique { it.id }
+        switchHelper.synchronizeAndCollectFixedDiscrepancies(involvedSwitchesforBothFlows).isEmpty()
 
         and: "Both flows are pingable"
-        [qinqFlow.flowId, vlanFlow.id].each {
-            verifyAll(northbound.pingFlow(it, new PingInput())) {
+        [qinqFlow, vlanFlow].each {
+            verifyAll(it.ping()) {
                 it.forward.pingSuccess
                 it.reverse.pingSuccess
             }
         }
 
         then: "Both flows allow traffic"
         if(!trafficDisclaimer) {
-            def examSimpleFlow = new FlowTrafficExamBuilder(topology, traffExam)
-                    .buildBidirectionalExam(vlanFlow, 1000, 5)
-            withPool {
-                [examQinQFlow.forward, examQinQFlow.reverse, examSimpleFlow.forward, examSimpleFlow.reverse]
-                        .eachParallel { direction ->
-                            def resources = traffExam.startExam(direction)
-                            direction.setResources(resources)
-                            assert traffExam.waitExam(direction).hasTraffic()
-                        }
-            }
+            def examSimpleFlow = vlanFlow.traffExam(traffExam, 1000, 5)
+            verifyFlowHasBidirectionalTraffic(examQinQFlow, traffExam)
+            verifyFlowHasBidirectionalTraffic(examSimpleFlow, traffExam)
         }
 
         when: "Update the QinQ flow(outer/inner vlans)"
-        def updateResponse = flowHelperV2.updateFlow(qinqFlow.flowId, qinqFlow.tap {
-            qinqFlow.source.vlanId = vlanFlow.source.vlanId
-            qinqFlow.source.innerVlanId = vlanFlow.destination.vlanId
-            qinqFlow.destination.vlanId = vlanFlow.destination.vlanId
-            qinqFlow.destination.innerVlanId = vlanFlow.source.vlanId
-        })
+        def updateQinqFlowEntity = qinqFlow.tap {
+            it.source.vlanId = vlanFlow.source.vlanId
+            it.source.innerVlanId = vlanFlow.destination.vlanId
+            it.destination.vlanId = vlanFlow.destination.vlanId
+            it.destination.innerVlanId = vlanFlow.source.vlanId
+        }
+        def updatedQinqFlow = qinqFlow.update(updateQinqFlowEntity)
```

Comment:
update() includes waiting for flow to be in UP state

Reply:
Done, fixed to sendUpdateRequest()